templates/vanilla/packages/client/src/mud/setup.ts

```diff
@@ -1,13 +1,45 @@
+/*
+ * This file sets up all the definitions required for a MUD client.
+ */
+
 import { createClientComponents } from "./createClientComponents";
 import { createSystemCalls } from "./createSystemCalls";
 import { setupNetwork } from "./setupNetwork";
 
+/*
+ * The type definition for the return type of setup.
+ * The result is an Awaited (https://www.typescriptlang.org/docs/handbook/utility-types.html#awaitedtype),
+ * which means a result that may not be immediately available.
+ *
+ * The Awaited result is of type ReturnType<typeof setup>, which
+ * means that TypeScript will see the type that the setup function
+ * returns and use that.
+ */
 export type SetupResult = Awaited<ReturnType<typeof setup>>;
 
+/*
+ * This is the setup function that is called from the client code
+ * (index.ts or index.tsx, depending on the client framework).
```

Comment:
since this is a comment in the client, can we make this specific to the client? or better yet, not reference it at all, so this isn't out of date if we move where it's called